pkg/tfbridge/schema.go

```diff
@@ -1866,3 +1883,24 @@ func ExtractInputsFromOutputs(oldInputs, outs resource.PropertyMap,
 	// Otherwise, take a schema-directed approach that fills out all input-only properties.
 	return extractSchemaInputsObject(outs, tfs, ps), nil
 }
+
+func getAssetTable(m resource.PropertyMap, ps map[string]*SchemaInfo, tfs shim.SchemaMap) (AssetTable, error) {
+	assets := AssetTable{}
+	val := resource.NewObjectProperty(m)
+	_, err := propertyvalue.TransformPropertyValue(
+		resource.PropertyPath{},
+		func(p resource.PropertyPath, v resource.PropertyValue) (resource.PropertyValue, error) {
+			if v.IsAsset() || v.IsArchive() {
+				schPath := PropertyPathToSchemaPath(p, tfs, ps)
+				info := LookupSchemaInfoMapPath(schPath, ps)
+				if info == nil {
+					return v, fmt.Errorf("unexpected asset %s", p.String())
```

Comment:
contract.Assertf here with descriptive error? This sounds like always a bridge invariant failure we need to fix in the bridge, not something caller or user can recover from, WDYT?

If you address this then getAssetTable does not need to return `error`.